### PR TITLE
Better sdp handling

### PR DESCRIPTION
**Pull request by rowleya**
**This pull request is not ready for merging**

This now includes two things:
1. Improvement in the SDP handling by not having timeouts
2. Addition of a "Big Data" protocol for handling large UDP packets *to the Ethernet chip only*

The core issue was fixed in a different branch.
This pull request remains open mainly as a reminder about the big data packet issue.

### CODE REVIEW

File: sark/sark_base.c [hunk 1]
```diff
@@ -508,56 +473,47 @@ void __attribute__((weak)) sark_post_main(void)
...
-    if (shm_msg == NULL) {
-        return 0;
-    }
-
-    sark_msg_cpy(shm_msg, msg);
-
-    sark.vcpu->mbox_mp_msg = shm_msg;
-    sark.vcpu->mbox_mp_cmd = SHM_MSG;
-
+static void set_msg_flag() {
```
Comment (dkfellows) [inline sark/sark_base.c]: Should be `(void)` to avoid weird edge case in C language spec.

**Comment by lplana** [comment]
@rowleya: what's the status of this pull request? I'm not sure if it was superseded by other SDP-related pull requests. I can review it if still pending.

**Comment by rowleya** [comment]
I was trying to do too many things here I think!  I think we have fixed the core issue in another branch, so this is here to remind me about the other things (mainly big data packets).

**Comment by lplana** [comment]
OK, then. I'll update the comment at the top - otherwise I'll keep asking the same question over and over again. Poor memory and all that.

**Comment by rowleya** [comment]
Looks so conflicted this would need redoing!

**Pull request closed without merging**